feat: claude integration

danorlando opened the pull request:
﻿Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.



## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- [X] Conversations with all models with title generation
- [X] Editing settings, saving, editing, exporting and recalling presets 
- [X] Aborting requests
- [X] Tested formatting of code generation, lists, etc.
- Note: reverse proxy is implemented, but has not been tested

### **Test Configuration**:
You'll need to set an `ANTHROPIC_API_KEY` in the .env. (or leave set to user_provided)

Note that documentation has not yet been updated to reflect the adding of this feature.

danorlando committed:
feat: bare bones implementation of claude client (WIP)

danorlando committed:
feat: bare bones implementation of claude client (WIP)

danorlando committed:
feat: client implementation of Claude (WIP)

danorlando committed:
fix: add claude to store

danorlando committed:
switch eventsource

kuangxiaoye commented:
When will Claude's work be merged into the main branch?(✧∀✧)

danorlando committed:
Try new method of calling claude with anthropic sdk

danorlando committed:
(WIP) Finish initial claude client implementation and api

danorlando committed:
debugging update

danny-avila commented on client/public/assets/microsoft.png:
look for SVGs, i think we can find some for both of these new assets

fuegovic replied:
I couldn't find the claude one so I made it
the microsoft one was easy to find
[svg assets.zip](https://github.com/danny-avila/LibreChat/files/12024140/svg.assets.zip)


![claude](https://github.com/danny-avila/LibreChat/assets/32828263/70415bb0-7708-4a44-96cb-4cf7e30ec8a2)

![microsoft](https://github.com/danny-avila/LibreChat/assets/32828263/92706594-7392-43a2-9bb9-ee820f6bd6bb)

fuegovic replied:
and here's the PaLM logo in svg 
<img src="https://github.com/danny-avila/LibreChat/assets/32828263/7d84a845-6448-4565-8db7-86056c3d32fb" width="200">

danny-avila committed:
fix(ClaudeClient.js): fix prompt prefixes for HUMAN_PROMPT and AI_PROMPT
fix(ClaudeClient.js): refactor buildMessages logic for correct handling of messages
refactor(ClaudeClient.js): refactor buildPrompt method to buildMessages for use in BaseClient sendMessage method
refactor(ClaudeClient.js): refactor getCompletion method to sendCompletion for use in BaseClient sendMessage method
refactor(ClaudeClient.js): omit getMessageMapMethod method for future refactoring
refactor(ClaudeClient.js): remove unused sendMessage method to prefer BaseClient message
fix(askClaude.js): error in getIds method was causing a frontend crash, userMessage was not defined
fix(askClaude.js): import abortMessage function from utils module
feat(askClaude.js): add /abort route to handle message abort requests
feat(askClaude.js): create abortControllers map to store abort controllers
feat(askClaude.js): implement abortAsk function to handle message abort logic
feat(askClaude.js): add onStart callback to handle message start logic
feat(HoverButtons.jsx): add 'claude' as a supported endpoint for branching

danny-avila committed:
fix(ClaudeClient.js): update defaultPrefix and promptPrefix messages

includes 'Remember your instructions' as Claude is trained to recognize labels preceding colons as participants of a conversation

danny-avila commented:
> When will Claude's work be merged into the main branch?(✧∀✧)

it's nearing completion now

danorlando committed:
Change name from claude to anthropic

danorlando committed:
add settings to handleSubmit and models to endpoints

danorlando committed:
Implement Claude settings

danorlando committed:
use svg for anthropic icon

danorlando committed:
Implement abort

danorlando committed:
Implement reverse proxy

danorlando committed:
remove png icons

danorlando committed:
replace web browser plugin

danorlando committed:
remove default prefix

danorlando committed:
fix styling of claude icon

danorlando committed:
fix console error from svg properties

danorlando committed:
remove single quote requirement from eslintrc

danny-avila committed:
fix(AnthropicClient.js): fix labels for HUMAN_PROMPT and AI_PROMPT
feat(AnthropicClient.js): add support for custom userLabel and modelLabel options
feat(AnthropicClient.js): add user_id metadata to requestOptions in getCompletion method
feat(anthropic, AnthropicClient.js): add debug logging

danny-avila committed:
refactor(AnthropicClient.js): change promptSuffix variable declaration from let to const

danny-avila committed:
fix(EndpointOptionsDialog.jsx): remove unnecessary code that changes endpointName from 'anthropic' to 'Claude'
fix(utils/index.jsx): fix alternateName value for 'anthropic' from 'Claude' to 'Anthropic'

danny-avila committed:
fix(AnthropicIcon): fix sizing/rendering/name of anthropic icon

danny-avila committed:
fix(AnthropicClient.js): change maxContextTokens default value to 99999
fix(AnthropicClient.js): change maxResponseTokens default value to 1500
fix(AnthropicClient.js): remove unnecessary code for setting maxContextTokens and maxResponseTokens based on modelOptions
fix(AnthropicClient.js): change max_tokens_to_sample default value to 1500
fix(anthropic.js): pass endpointOption.token to AnthropicClient constructor

danorlando committed:
Update .env.example

danny-avila committed:
fix(AnthropicClient.js): remove exceeding message when it puts us over the token limit
fix(AnthropicClient.js): handle case when the first message exceeds the token limit
fix(AnthropicClient.js): throw error when prompt is too long
fix(AnthropicClient.js): adjust max tokens calculation to use maxOutputTokens
fix(anthropic.js): remove console.log statement in ask route

danny-avila committed:
Merge branch 'feat-claude-integration' of https://github.com/danorlando/librechat into feat-claude-integration

danny-avila committed:
feat(server/index): increase incoming json payload allowed size
